### PR TITLE
Carousel: Fix full image not being served by photon

### LINKED ISSUE #2053: Jetpack Carousel & Photon combination causes images to be upscaled
Normally when using the Jetpack Carousel, it will display the images in their original size. If they're larger than the browser window, they'll just be resized by the browser. While not ideal, it's better than what Photon does. After activating Photon, I noticed some images suddenly looked really blurry in the Carousel. This only happened with large images (1080P). What apparently happens is that Photon checks if the resolution of the original image is greater than the browser window. If so, it returns an image in the "Large" image setting. This is less than ideal, because the Large image resolution can be a lot smaller than the original image size. The browser then upscales the image, which looks rather terrible. In my opinion, Photon should never serve an image that's smaller than the resolution required for viewing in the Carousel, to avoid it being upscaled. After all, that's the standard behavior for the Carousel when not using Photon.

### LINKED ISSUE #2969: Carousel + Tiled Galleries loads full size image
I think this just started happening... if you have Carousel and Tiled Galleries activated, the full sized image is being loaded when you click on a thumbnail. Previously I believe the "large" size would load in.

Also, the full size image that loads when you click a thumbnail is not being served by Photon (if it's activated).

### PR DESCRIPTION
**Pull request by Draikin**
Fixes #2969

This adds an additional check to determine whether or not the carousel needs to return the full image served by Photon. Before, photon was never used to serve the original image.

### CODE REVIEW

**Comment by eliorivero** [comment]
good to go :100:

File: modules/carousel/jetpack-carousel.js [hunk 1]
```diff
@@ -987,6 +987,17 @@ jQuery(document).ready(function($) {
 			if ( medium_width >= args.max_width || medium_height >= args.max_height ) {
 				return args.medium_file;
 			}
+			
+			if ( isPhotonUrl ) {
+				// args.orig_file doesn't point to a Photon url, so in this case we use args.large_file
+				// to return the photon url of the original image.
+				var largeFileIndex = args.large_file.lastIndexOf( '?' );
+				var origPhotonUrl = args.large_file;
+				if ( largeFileIndex != -1 ) {
```
Comment (dereksmart) [inline modules/carousel/jetpack-carousel.js]: can you change this to strict operator `!==`?  our js tests are picky. 

Also please use <a href="https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#yoda-conditions">Yoda style conditional</a> for this.  Thanks!

**Comment by Draikin** [comment]
I made the requested changes for the coding style.

**Comment by dereksmart** [comment]
Thank you @Draikin.  Looks like the extra lines of code you fixed for standards are now causing merge conflicts :( can you please fix?

**Comment by eliorivero** [comment]
lgtm
Test and code were ok and now `grunt` ends without errors.

**Pull request merged**